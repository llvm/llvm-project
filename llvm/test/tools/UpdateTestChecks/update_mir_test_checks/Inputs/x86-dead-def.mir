# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
# RUN: llc -mtriple=x86_64-linux-gnu -run-pass=none -verify-machineinstrs %s -o - | FileCheck %s --check-prefixes=CHECK

--- |

  define void @test() {
    unreachable
  }

...
---
name:            test
alignment:       16
registers:
  - { id: 0, class: gpr, preferred-register: '' }
body:             |
  bb.1:
    dead %0(s32) = G_CONSTANT i32 0

...
