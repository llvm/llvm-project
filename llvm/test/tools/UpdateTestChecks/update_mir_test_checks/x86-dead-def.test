# REQUIRES: x86-registered-target
## Check that update_mir_test_checks replaces vregs in dead definitions

# RUN: cp -f %S/Inputs/x86-dead-def.mir %t.mir && %update_mir_test_checks %t.mir
# RUN: diff -u %S/Inputs/x86-dead-def.mir.expected %t.mir
