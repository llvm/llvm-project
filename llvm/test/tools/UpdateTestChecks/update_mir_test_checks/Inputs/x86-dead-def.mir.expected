# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
# RUN: llc -mtriple=x86_64-linux-gnu -run-pass=none -verify-machineinstrs %s -o - | FileCheck %s --check-prefixes=CHECK

--- |

  define void @test() {
    unreachable
  }

...
---
name:            test
alignment:       16
registers:
  - { id: 0, class: gpr, preferred-register: '' }
body:             |
  bb.1:
    ; CHECK-LABEL: name: test
    ; CHECK: dead [[C:%[0-9]+]]:gpr(s32) = G_CONSTANT i32 0
    dead %0(s32) = G_CONSTANT i32 0

...
